feature_engine/selection/smart_correlation_selection.py

```diff
@@ -143,11 +146,27 @@ def __init__(
         cv=3,
     ):
 
-        if method not in ["pearson", "spearman", "kendall"]:
+        if method not in ["pearson", "spearman", "kendall"] and not callable(method):
             raise ValueError(
-                "correlation method takes only values 'pearson', 'spearman', 'kendall'"
+                "correlation method takes only values 'pearson', 'spearman', "
+                + "'kendall' or callable."
             )
 
+        # check callable method takes two input arguments
+        if callable(method):
+            sig = signature(method)
+            positional_or_keyword_args = sum(
```

Comment:
Perhaps: 

```
positional_or_keyword_args = [param for param in sig.paramters.values() if param.kind == param.POSITIONAL_OR_KEYWORD]

if len(positional_or_keyword_args) != 2:
    ...
```

Reply:
Also (sorry, I'm not that familiar with the context), do we want to catch callables with keyword args?

e.g.
```
def foo(a, *, b:int):
    pass
```
we wouldn't catch `b` with the above check, we'd need to also check for kind `KEYWORD_ONLY`

Reply:
@ChristopherGS  you're right we wouldn't be catching also 

```python
def foo2(*, a:int, b:int):
    pass
```

or catching more arguments than needed.

```
def foo3(x, y, *, z=1, w='something'):
    pass
```

the idea behind this check is to avoid a `TypeError` when calling the method as it only expects two input arguments. For example
```
def corrf(x):
    pass

corrf(1, 2)
>>> TypeError: foo4() takes 1 positional argument but 2 were given
```
 
I think it could be better if the error is directly raise by `pandas.corr()`. What are your thoughts ?

Reply:
> I think it could be better if the error is directly raise by `pandas.corr()`. What are your thoughts ?

agree

Reply:
OK, so just to summarize, we will remove this functionality from the code. If anything else, for my clarity lol

Reply:
ok, so we remove this from `DropCorrelatedFeatures` and `SmartCorrelatedSelection` and the error is raised by pandas when calling `fit` or `fit_transform`.